### PR TITLE
Update flatbuffers in workspace.bzl

### PR DESCRIPTION
**Pull request by Saduf2019**
Updated Flatbuffers to "2.0" Uas dependency change to "https://github.com/tensorflow/tensorflow/pull/51504"

### CODE REVIEW

File: third_party/flatbuffers/workspace.bzl [hunk 1]
```diff
@@ -5,7 +5,7 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo():
     tf_http_archive(
         name = "flatbuffers",
-        strip_prefix = "flatbuffers-1.12.0",
+        strip_prefix = "flatbuffers-2.0",
         sha256 = "62f2223fb9181d1d6338451375628975775f7522185266cd5296571ac152bc45",
         urls = [
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v1.12.0.tar.gz",
```
Comment (lgeiger) [inline third_party/flatbuffers/workspace.bzl:11]: The download URLs and hash, will also require an update, otherwise the build will fail.

**Review by lgeiger — CHANGES_REQUESTED** [review]

File: third_party/flatbuffers/workspace.bzl (hunk 1 above)
Comment (mihaimaruseac) [inline third_party/flatbuffers/workspace.bzl:11]:
```suggestion
            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v2.0.0.tar.gz",
```

File: third_party/flatbuffers/workspace.bzl [hunk 2]
```diff
@@ -5,10 +5,10 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo():
     tf_http_archive(
         name = "flatbuffers",
-        strip_prefix = "flatbuffers-1.12.0",
+        strip_prefix = "flatbuffers-2.0",
         sha256 = "62f2223fb9181d1d6338451375628975775f7522185266cd5296571ac152bc45",
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v1.12.0.tar.gz",
+            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/flatbuffers/archive/v2.0.0.tar.gz",
             "https://github.com/google/flatbuffers/archive/v1.12.0.tar.gz",
```
Comment (mihaimaruseac) [inline third_party/flatbuffers/workspace.bzl:12]:
```suggestion
            "https://github.com/google/flatbuffers/archive/v2.0.0.tar.gz",
```

File: third_party/flatbuffers/workspace.bzl [hunk 3]
```diff
@@ -5,11 +5,11 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo():
     tf_http_archive(
         name = "flatbuffers",
-        strip_prefix = "flatbuffers-1.12.0",
+        strip_prefix = "flatbuffers-2.0",
         sha256 = "62f2223fb9181d1d6338451375628975775f7522185266cd5296571ac152bc45",
```
Comment (mihaimaruseac) [inline third_party/flatbuffers/workspace.bzl:9]:
```suggestion
        sha256 = "9ddb9031798f4f8754d00fca2f1a68ecf9d0f83dfac7239af1311e4fd9a565c4",
```

**Review by mihaimaruseac — APPROVED** [review]

File: third_party/flatbuffers/workspace.bzl [hunk 4]
```diff
@@ -5,11 +5,11 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo():
     tf_http_archive(
         name = "flatbuffers",
-        strip_prefix = "flatbuffers-1.12.0",
-        sha256 = "62f2223fb9181d1d6338451375628975775f7522185266cd5296571ac152bc45",
+        strip_prefix = "flatbuffers-2.0",
```
Comment (mihaimaruseac) [inline third_party/flatbuffers/workspace.bzl:8]:
```suggestion
        strip_prefix = "flatbuffers-2.0.0",
```

**Review by mihaimaruseac — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by lu-wang-g** [comment]
Flatbuffer 2.0 has some incompatible changes that we need to resolve from internally first. It will be difficult to debug and test from external contribution. Therefore I'll reject this PR for now, and we'll do the upgrade later.

**Pull request closed without merging**